portfolio/src/main/java/com/google/sps/servlets/BookUploadServlet.java

```diff
@@ -0,0 +1,52 @@
+package com.google.sps.servlets;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.sps.data.Book;
+import com.google.sps.data.BookReader;
+import com.google.sps.data.BookUploadUtility;
+import java.io.IOException;
+import java.util.Map;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+// Initiate upload books into the books datastore
+@WebServlet("/admin-book-upload")
+public class BookUploadServlet extends HttpServlet {
+  private final BookUploadUtility bookUploadUtility = new BookUploadUtility();
+  private Map<Integer, Book> bookList;
+
+  @Override
+  public void init() throws ServletException {
+    try {
+      BookReader reader = new BookReader(getServletContext().getRealPath("/WEB-INF/20_books.csv"));
```

Comment:
are we not using all_books.csv now?

Reply:
I wanted to start with the 20 books to get it working first. Once that is working we can reset the datastore and use all the books.

Reply:
Got it!